src/classes/ADDR_Seasonal_TEST.cls

```diff
@@ -0,0 +1,923 @@
+        from Account]);
+        
+        for (Account acc : mapAccIdAcc.values()) {
+            system.assert(acc.BillingStreet.contains('New Seasonal Street'));
+            system.assert(acc.BillingCity.contains('New Seasonal City'));
+            system.assertEquals(false, acc.is_Address_Override__c);
+            system.assertNotEquals(null, acc.Current_Address__c);
+        }
+        
+        // verify the previous addresses still are Default
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, Parent_Account__c, Latest_Start_Date__c, Latest_End_Date__c from Address__c];
+        system.assertEquals(2 * 2, listAddr.size());
+        for (Address__c addr : listAddr) {
+            boolean fSeasonal = (addr.MailingStreet__c.contains('New Seasonal Street'));
+            system.assertEquals(!fSeasonal, addr.Default_Address__c);
+        }        
+    }
+    
+    /*********************************************************************************************************
+    @description
+        insert new seasonal addresses to HH w/ existing default addresses  
+    verify:
+        contact's && hh address matches new seasonal
+        old default addresses still marked default
+    **********************************************************************************************************/            
+    @isTest
+    public static void insertNewSeasonalAddrWithYear() {
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+
+        // create additional addresses
+        addrs = UTIL_UnitTestData_TEST.initTestAddr(2);
+        for (Integer i = 0; i < 2; i++) {
+            addrs[i].Parent_Account__c = accsAddrs.accs[i].Id;
+            addrs[i].Default_Address__c = false;
+            
+            addrs[i].Seasonal_Start_Year__c = System.today().addYears(-1).year();
+            addrs[i].Seasonal_Start_Month__c = String.valueOf(system.today().addMonths(-1).month());
+            addrs[i].Seasonal_Start_Day__c = '1';
+            
+            addrs[i].Seasonal_End_Year__c = System.today().addYears(1).year();
+            addrs[i].Seasonal_End_Month__c = String.valueOf(system.today().addMonths(1).month());
+            addrs[i].Seasonal_End_Day__c = '28';
+            
+            addrs[i].MailingStreet__c = 'New Seasonal Street' + i;
+            addrs[i].MailingCity__c = 'New Seasonal City' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.stopTest();
+        
+        // verify Account address is new!
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.